app/notification.go

```diff
@@ -438,7 +438,11 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 	message := model.NewWebSocketEvent(model.WebsocketEventPosted, "", post.ChannelId, "", nil)
 
 	// Note that PreparePostForClient should've already been called by this point
-	message.Add("post", post.ToJson())
+	postJSON, jsonErr := post.ToJSON()
+	if jsonErr != nil {
+		return nil, errors.Wrapf(jsonErr, "failed to encode post to JSON")
+	}
+	message.Add("post", postJSON)
```

Comment:
Definitely, we need to use `json.RawMessage` here. But that requires minor client side changes, because the client re-parses each prop object again.

This is something that has bothered me for long, and now we really have a chance to nail this. (Not required for v6 of course).

@hmhealey - wondering if you can chime in on the amount of effort required on the client side to remove all the unnecessary `JSON.parse` calls for each prop object?

Reply:
In terms of just code changes, it's probably not a horrible amount of work, but I'd be worried about us missing places that it might be needed. The websocket handling on the client hasn't even been migrated to TypeScript yet, so we don't even have that much protection.

I'd also be worried about older client compatibility since changing that would almost certainly break mobile/Matterhorn/anyone else using websockets. I feel like we'd need a way to version the websocket API to be able to change that.

Reply:
Fair point.